locations/spiders/bc_pizza.py

```diff
@@ -1,95 +1,136 @@
-    ]
-    custom_settings = {"ROBOTSTXT_OBEY": False}
-
-    def convert_hours(self, hours):
-        hours = [x.strip() for x in hours]
-        hours = [x for x in hours if x]
-        for i in range(len(hours)):
-            converted_times = ""
-            if hours[i] != "Closed":
-                if hours[i] != "Open 24 Hours":
-                    from_hr, to_hr = (hr.strip() for hr in hours[i].split("-"))
-                    if re.search(regex_am, from_hr):
-                        from_hr = re.sub(regex_am, "", from_hr)
-                        hour_min = from_hr.split(":")
-                        if len(hour_min[0]) < 2:
-                            hour_min[0].zfill(2)
-                        converted_times += (":".join(hour_min)) + "-"
-                    else:
-                        from_hr = re.sub(regex_pm, "", from_hr)
-                        hour_min = from_hr.split(":")
-                        if int(hour_min[0]) < 12:
-                            hour_min[0] = str(12 + int(hour_min[0]))
-                        converted_times += (":".join(hour_min)) + "-"
-
-                    if re.search(regex_am, to_hr):
-                        to_hr = re.sub(regex_am, "", to_hr)
-                        hour_min = to_hr.split(":")
-                        if len(hour_min[0]) < 2:
-                            hour_min[0].zfill(2)
-                        if int(hour_min[0]) == 12:
-                            hour_min[0] = "00"
-                        converted_times += ":".join(hour_min)
-                    else:
-                        to_hr = re.sub(regex_pm, "", to_hr)
-                        hour_min = to_hr.split(":")
-                        if int(hour_min[0]) < 12:
-                            hour_min[0] = str(12 + int(hour_min[0]))
-                        converted_times += ":".join(hour_min)
-                else:
-                    converted_times = "00:00-24:00"
-            else:
-                converted_times += "off"
-            hours[i] = converted_times
-        days = ["Mo", "Tu", "We", "Th", "Fr", "Sa", "Su"]
-        hours = "".join("{} {} ".format(*t) for t in zip(days, hours))
-        return hours
+    start_urls = ["https://bc.pizza/project-cat/bc-pizza-locations/"]
 
     def parse(self, response):
-        results = response.json()
-        for i in results:
-            ref = i["id"]
-            street = i["address"]
-            city = i["city"]
-            state = i["state"]
-            postcode = i["zip"]
-            lat = float(i["lat"])
-            lon = float(i["lng"])
-            hours = i["hours"]
-            hours = self.convert_hours(re.findall(regex_hours, hours))
-            phone = i["phone"]
-            website = i["facebook_url"]
-            addr_full = "{} {}, {} {}".format(street, city, state, postcode)
+        urls = response.xpath("//a[@rel='bookmark']/@href").extract()
+        for url in urls:
+            yield scrapy.Request(url, self.parse_locations)
```

Comment:
This one could be a [sitemap spider](https://bc.pizza/portfolio-sitemap.xml), or a [CrawlSpider](https://docs.scrapy.org/en/latest/topics/spiders.html#crawlspider). Just a recommendation from me though, not a requirement because of how simple the current method is.